clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1199,7 +1199,7 @@ public ClientRequest newClientRequest(String nodeId,
                                           AbstractRequest.Builder<?> requestBuilder,
                                           long createdTimeMs,
                                           boolean expectResponse) {
-        return newClientRequest(nodeId, requestBuilder, createdTimeMs, expectResponse, defaultRequestTimeoutMs, null);
+        return newClientRequest(nodeId, requestBuilder, createdTimeMs, expectResponse, defaultRequestTimeoutMs, null, "");
```

Comment:
I think it would be best to use null here as the "not set" value.  We never clearly articulated that the empty string is not valid for principals.